quantumdb-core/src/main/java/io/quantumdb/core/migration/operations/TransitiveTableMirrorer.java

```diff
@@ -35,7 +35,13 @@
 				continue;
 			}
 
-			String newRefId = RandomHasher.generateRefId(refLog);
+			String newRefId;
+			if (cleanup && !tableRef.getRefId().equals(tableRef.getName())) {
```

Comment:
If cleanup is true and the table does not already have it's original name, rename it to its original name, otherwise generate RefId